tracing.go

```diff
@@ -111,13 +128,38 @@ type publisherDecorator struct {
 func (d *publisherDecorator) Publish(topic string, messages ...*message.Message) error {
 	for i := range messages {
 		msg := messages[i]
+
 		span := trace.FromContext(msg.Context())
 		if span == nil {
 			d.logger.Debug("Span context nil, cannot propagate", watermill.LogFields{"topic": topic})
-		} else {
-			SetSpanContext(span.SpanContext(), msg)
+			continue
 		}
+
+		eID := generateEventID()
+		eIDString := strconv.FormatInt(eID, 10)
+		msg.Metadata.Set(spanEventIDKey, eIDString)
+
+		messageBytes := []byte(msg.Payload)
+		messageBytesSize := len(messageBytes)
+
+		// TODO!!! test it!
+		span.AddAttributes(trace.StringAttribute(spanEventPayloadAttributeKey, string(messageBytes)))
+		span.AddMessageSendEvent(eID, int64(messageBytesSize), 0)
+
+		SetSpanContext(span.SpanContext(), msg)
 	}
 
 	return d.Publisher.Publish(topic, messages...)
 }
+
+func generateEventID() int64 {
+	maxUint64 := ^uint64(0)
```

Comment:
If I'm not wrong there are constants defined in `math` module: `MaxUint64`, `MinInt64`, you can consider using them here.

Reply:
https://github.com/czeslavo/watermill-opencensus/pull/6/commits/0c281d9a162452456cbca755e765cb33add72009